code/modules/makeshift/makeshift_reagents.dm

```diff
@@ -0,0 +1,253 @@
+/obj/structure/chemkit
+	name = "makeshift workstation"
+	icon = 'icons/obj/makeshift_workstation.dmi'
+	icon_state = "workstation"
+	desc = "It's a makeshift workstation for grinding, chopping, and heating."
+	density = 1
+
+	anchored = 1
+	throwpass = 1
+
+	var/obj/item/device/analyzer/analyzer
+	var/transfer_out = 0
+	var/phase_filter
+
+	var/list/forbidden = list(/obj/item/weapon/reagent_containers/inhaler, /obj/item/weapon/reagent_containers/hypospray, /obj/item/weapon/reagent_containers/glass, /obj/item/weapon/extinguisher)
+	// duplicate from blender code, since it's not really worth a define. also, it has fewer things.
+	var/list/sheet_reagents = list( //have a number of reagents which is a factor of REAGENTS_PER_SHEET (default 20) unless you like decimals
+		/obj/item/stack/material/iron = list("iron"),
+		/obj/item/stack/material/uranium = list("uranium"),
+		/obj/item/stack/material/phoron = list("phoron"),
+		/obj/item/stack/material/gold = list("gold"),
+		/obj/item/stack/material/silver = list("silver"),
+		/obj/item/stack/material/steel = list("iron", "carbon"),
+		/obj/item/stack/material/sandstone = list("silicon", "acetone"),
+		/obj/item/stack/material/glass = list("silicate"),
+		) // removed borosilicate glass, platinum, and plasteel, too tough. just steal a grinder if you need it
+
+/obj/structure/chemkit/Initialize()
+	. = ..()
+	create_reagents(180)
+
+/obj/structure/chemkit/examine(mob/user)
+	. = ..()
+	if(analyzer)
+		to_chat(user, span("notice", "The analyzer displays that the temperature is [round(reagents.get_temperature() - T0C,0.1)]C."))
+
+/obj/structure/chemkit/verb/phase_filter()
+	set name = "Set Phase Filter"
+	set category = "Object"
+	set src in view(1)
+	if(!use_check(usr))
```

Comment:
```suggestion
	if(use_check_and_message(usr))
```

Reply:
No. `use_check` returns a nonzero error value if it fails.

Reply:
Oh, wait. I'm dumb. I read this backwards.